src/components/Layer.js

```diff
@@ -308,8 +309,17 @@ export class Layer extends Component {
 	 */
 	createLayer = async () => {
 		await this.updateData();
-		// TODO: Create new ukey for layer
-		this.props.addLayer(this.state);
+		const proposedLayer = {...this.state};
+
+		if (proposedLayer.name === NEW_LAYER.name) {
```

Comment:
Is there a reason to enforce a layer name? Not displayed yes, but I am afraid of future bugs if we set a property automatically. I would prefer using the `generateRandomName` inside the `generateUkeyFromName` if name is undefined. What do you think?

Reply:
Generating random name allows for visual differentiation between newly created layer, when both are collapsed. If anything, it's easily modifiable after layer creation. Alternatively, we could have validation on the layer forms, deactivating create/update buttons until the title (not very critical) and other required layer fields are filled in appropriately.
![image](https://user-images.githubusercontent.com/11217077/104608748-1dea0c80-5637-11eb-8390-d9424a08d10f.png)

Reply:
Btw, validation of form fields is critical future work. As an extreme example,  today you can just boot the app, expand new layer form and click to create a new layer without filling in any of the fields. This creates a (considered) valid layer which the map attempts to render, therefore crashing the app...

Reply:
It's the UI's job to prevent the user from shooting themselves in the foot :), and in that direction, we are just a few iterations away from great UX

Reply:
Users may not need label, or want not to have labels, there is no reason for us to enforce that at this point.

Regarding data validation, there is probably work to be done here, but in the worst case scenario, the default layer returns no data, which is something that the app should manage (right now, it may fail due to the layer bounds, but that's something to fix).

Reply:
Any thoughts on this? Otherwise please postpone the work on the "default" fields, let's not mix everything.

Reply:
I agree only added change here to opportunistically improve the shipped experience.

PS: I didn't get any free time in the recent days, and hope I'm not blocking on PR. Feel free to push changes and merge

Reply:
I can't be annoyed with people don't having free time ;-) I'll merge this one and fix the remaining small stuff in the target branch before pushing to master. Thank you!